lib/oxttl/src/lexer.rs

```diff
@@ -823,6 +834,93 @@ impl N3Lexer {
         Ok(Some(c))
     }
 
+    fn recognize_utf16_surrogate_pair(
+        data: &[u8],
+        position: usize,
+    ) -> Result<Option<char>, TokenRecognizerError> {
+        if data.len() < 4 {
+            return Ok(None);
+        }
+
+        let val_high = str_from_utf8(&data[..4], position..position + 6)?;
+        let surrogate_high = u16::from_str_radix(val_high, 16).map_err(|e| {
+            (
+                position..position + 6,
+                format!(
+                    "The escape sequence '\\u{val_high}' is not a valid hexadecimal string: {e}"
+                ),
+            )
+        })?;
+
+        // https://www.unicode.org/glossary/#low_surrogate_code_point
+        if matches!(surrogate_high, 0xDC00..=0xDFFF) {
+            return Err((
+                position..position + 6,
+                format!("UTF-16 low-surrogate escape sequence '\\u{val_high}' must be preceded by a high-surrogate"),
+            )
+                .into());
+        }
```

Comment:
Open question: do we care about validating this here? I believe`char::decode_utf16` should already fail in this case.